lib/MongoDB/MongoClient.pm

```diff
@@ -1524,13 +1522,18 @@ sub send_admin_command {
         ref($read_pref) ? $read_pref : ( mode => $read_pref ) )
       if $read_pref && ref($read_pref) ne 'MongoDB::ReadPreference';
 
+    # explicitly called as scalar. Calling inside a hash(ref) causes list
+    # context, giving an empty list and an odd length of array. Provided so can
+    # get the latest $clusterTime
+    my $session = $self->_start_implicit_session;
```

Comment:
This makes no sense to me as there is a singular return value (unless Moo's new() does something weird in list context?!?).  And if it were really needed, I'd rather see it wrapped in `scalar(...)` than using a temporary variable.    Can you explain to me what's going on?

Reply:
The issue is due to the return from _start_implicit_session - in scalar context, the implicit value is undef, but in list context it is an empty list. When inside the hash of new(), its in list context, returns an empty list, and then you end up with a hash with an odd number of elements. Could put `session => $self->_start_implicit_session || undef` in the new statement instead I suppose, but that would still need an explanation as to why so it doesnt get removed by accident later

Reply:
Oh, the shortcut return if not supported.  Github helpfully folded that line for me.  I'd prefer that you return undef explicitly in _start_implicit_session (with a `## no critic` comment).  That also makes me think that we should change the name for clarity: `s/_start_implicit_session/_maybe_get_implicit_session/`

Reply:
K, have refactored for that